File: basic/go/basic.go
package main

import (
	"fmt"
	"os"
	"time"
)

func main() {
	pid := os.Getpid()
	ppid := os.Getppid()

	argc := len(os.Args)
	fmt.Printf("  PID     %d\n", pid)
	fmt.Printf("  PPID    %d\n", ppid)
	fmt.Printf("  argc    %d\n", argc)
	for i := 0; i < argc; i++ {
		fmt.Printf("  argv[%d] %s\n", i, os.Args[i])
	}

	fmt.Print("  Sleeping for 60 seconds so you can lookup the PID")
	time.Sleep(60 * time.Second)
}

File: basic/nim/basic.nim
import os
import strformat
import posix

proc main() =
    echo(fmt"  PID     {getpid()}")
    echo(fmt"  PPID    {getppid()}")
    echo(fmt"  argc    {paramCount() + 1}")  # Nim doesn't count argv[0] in paramCount

    for i in 0..paramCount():
        echo(fmt"  argv[{i}] {paramStr(i)}")

    echo("  Sleeping for 60 seconds so you can lookup the PID")
    setControlCHook(proc() {.noconv.} = discard)
    sleep(60 * 1000)

main()
